### PR TITLE
Create TenetFarm.md

### PR DESCRIPTION
**Pull request by tenet-farm**
Grants for TENET

# Grant Application Checklist

- [ ] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [ ] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [ ] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [ ] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [ ] The initial PR contains only one commit (squash if needed before submitting your PR).
- [ ] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. However the current application is above the 30k limit of the grants program. Or what are the total costs of the application? Also could you provide more technical information (e.g. programming language) as part of the deliverables. Furthermore is you complete tech stack open source? For example, I couldn’t find you layer 2 solution on github: https://github.com/TenetFarm (but I haven’t looked into it in detail yet)

**Comment by tenet-farm** [comment]
> Thanks for the application. However the current application is above the 30k limit of the grants program. Or what are the total costs of the application? Also could you provide more technical information (e.g. programming language) as part of the deliverables. Furthermore is you complete tech stack open source? For example, I couldn’t find you layer 2 solution on github: https://github.com/TenetFarm (but I haven’t looked into it in detail yet)

Sorry for misunderstanding about the application, we have updated it. 

About technical side:
Smart Contract: Solidity
Frontend: ReactJS
Backend: C++

And till now, not all the tech stack open sourced. Only the smart contract part is opened.

**Comment by Noc2** [comment]
Thanks for the update. Could you integrate this information into the application and also fill out the Overview part of the development roadmap? Are you planning to open source the other parts and if so could you integrate this into the deliverables?

**Comment by tenet-farm** [comment]
> Thanks for the update. Could you integrate this information into the application and also fill out the Overview part of the development roadmap? Are you planning to open source the other parts and if so could you integrate this into the deliverables?

Thanks for your review. We have updated the Overview part and integrate the technical information into the application. For the front-end and back-end codes, we will not open it now, because all the functions of tenet can be called via smart contracts which already open sourced.

File: applications/TenetFarm.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+* **FTE:**  2
+* **Costs:** 0.8 BTC
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | Tenet documentation center (Help center) is live, we will provide more inline documentation of the code and a basic tutorial to show how Tenet (Polkadot version) works. |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. |  
+| 1. | Chain Adapter | Has already adapted Polkadot, ETH and BSC|
```
Comment (Noc2) [inline applications/TenetFarm.md:136]: I was mostly referring to the technical details of these deliverables. What’s exactly the adapter and how is it implemented (programming language etc.)?

File: applications/TenetFarm.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | Tenet documentation center (Help center) is live, we will provide more inline documentation of the code and a basic tutorial to show how Tenet (Polkadot version) works. |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. |  
+| 1. | Chain Adapter | Has already adapted Polkadot, ETH and BSC|  
+| 2. | Create Liquidity Tap | Has supported creating Liquidity Tap for AMM platform on Polkadot |  
+| 3. | Trading Aggregator | Has aggregated the AMM platforms on Polkadot and completed the development of trading aggregator |
```
Comment (Noc2) [inline applications/TenetFarm.md:138]: Same for 2 and 3

**Review by Noc2 — COMMENTED** [review]
Thanks for updating the application. Could you provide more information regarding my comments below. Also something you should be aware of in general, we usually don’t fund the development of solidity smart contracts via our grants program. Simply because we are no experts when it comes to solidity smart contracts and the chances are pretty high that the code already exists. I think it might make more sense for you to potentially sent an application to the moonbeam grants program: https://moonbeam.network/community/grants/

File: applications/TenetFarm.md (hunk 2 above)
Comment (tenet-farm) [reply applications/TenetFarm.md]: Updated

File: applications/TenetFarm.md (hunk 1 above)
Comment (tenet-farm) [reply applications/TenetFarm.md]: Updated

**Comment by tenet-farm** [comment]
> Thanks for updating the application, but I was referring to the deliverables of the milestone. Sorry for the misunderstanding. Could you provide more information regarding my comments below. Also something you should be aware of in general, we usually don’t fund the development of solidity smart contracts via our grants program. Simply because we are no experts when it comes to solidity smart contracts and the chances are pretty high that the code already exists. I think it might make more sense for you to potentially sent an application to the moonbeam grants program: https://moonbeam.network/community/grants/

Yes, most of our features now are developed in Solidity. We have talked with Patract team and some Parity engineers in the past few days, our plan now is implementing the core tenet features and bridge on substrate via solidity in the 1st milestone, and in the next milestone, we will build the new features in INK. And we will find some chance to talk with MoonBeam team as well.

**Comment by mmagician** [comment]
@tenet-farm Given your latest response, I'll close this PR for now. Once you have the first part of the developments ready, feel free to re-open the application.

**Pull request closed without merging**